kong/clustering/tls.lua

```diff
@@ -230,4 +232,25 @@ function tls.validate_client_cert(kong_config, cp_cert, dp_cert_pem)
 end
 
 
+--- Extract certificate details from the data plane certificate.
+---
+---@param dp_cert_pem string    # data plane cert text
+---
+---@return table? cert_details  # certificate details
+function tls.extract_dp_cert(dp_cert_pem)
+  local cert, err = openssl_x509.new(dp_cert_pem, "PEM")
```

Comment:
Please avoid repeated parsing of the PEM cert in each connection, it is expensive.

Try to extract this info inside `validate_client_cert` and return at the same time.

Reply:
update